packages/react-instantsearch-hooks/src/hooks/useConnector.ts

```diff
@@ -14,7 +14,8 @@ export function useConnector<
   TDescription extends WidgetDescription
 >(
   connector: Connector<TDescription, TProps>,
-  props: TProps = {} as TProps
+  props: TProps = {} as TProps,
+  widgetType: string | undefined
```

Comment:
We could rather pass the object signature as an object (`{ $$widgetType }`) and spread it on the widget instance instead of just the string widget type. Here, when we need to introduce something new, it will force us to introduce yet another argument, with this one being able to be undefined.

Reply:
Yes, you're right, I copied from the WIP branch instead of the RFC; good catch!